tests/python_package_test/test_basic.py

```diff
@@ -579,3 +580,32 @@ def test_param_aliases():
     assert all(len(i) >= 1 for i in aliases.values())
     assert all(k in v for k, v in aliases.items())
     assert lgb.basic._ConfigAliases.get('config', 'task') == {'config', 'config_file', 'task', 'task_type'}
+
+
+def _bad_gradients(labels, preds):
+    return np.random.randn(len(labels) + 1), np.random.rand(len(labels) + 1)
```

Comment:
Sorry for the oversight in my previous review! Current signature is for sklearn wrapper, standard API accepts Dataset object instead of true labels.
https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.Booster.html#lightgbm.Booster.update

```suggestion
def _bad_gradients(preds, train_data):
    return np.random.randn(len(preds) + 1), np.random.rand(len(preds) + 1)
```

Reply:
2f6b647